### ISSUE TITLE
[Bug]: Failed to delete data resulting in repeated insertion of data(删除数据未成功导致重复插入数据)

### ISSUE THREAD

**Issue by uniqueSkeeter** [milvus-io/milvus#38712]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:V2.40
- Deployment mode(standalone or cluster):standalone
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.4
- OS(Ubuntu or CentOS): Ubuntu
```

### Current Behavior

Failed to delete data resulting in repeated insertion of data

code as below:

# 先删除已有的商品
        for product in param.productList:
            query_filter = f"display_prd_id=={product.disPlayPrdId}"
            product_kb.delete(expr=query_filter)
        # 将商品信息转为Document
        docs = to_product_docs(param.productList)
        text_splitter = RecursiveCharacterTextSplitter(chunk_size=600, chunk_overlap=0)
        split_docs = text_splitter.split_documents(docs)
        # 文档向量化，并创建数据库
        product_kb.add_documents(split_docs)

### Anything else?

![20241224-172604](https://github.com/user-attachments/assets/dfc6fbed-fb41-4811-b4da-eca7df907b32)

**Comment by yanliang567** [comment]
@uniqueSkeeter quick questions:
1. is the field `disPlayPrdId` primary key? could you please share the schema? how do you ensure it is unique?
2. what is the Count:5 means in the attached screenshots. could you please share the code snippet about how to tell that the deletion is failed?

/assign @uniqueSkeeter 
/unassign

**Comment by uniqueSkeeter** [comment]
yes,  "disPlayPrdId"  is my logical primary key （but not table's real primary key ）. and as code see, I add data always delete by disPlayPrdId  then  insert a new column

![20241226-144120](https://github.com/user-attachments/assets/f190a62b-b312-40a0-9a59-de90a303cf3f)
![20241226-144253](https://github.com/user-attachments/assets/73e92372-2ef3-473e-850f-0a53dfd38308)

**Comment by yanliang567** [comment]
could u please share the code snippets of insert and query? i need to you how you define the deletion failure

**Comment by uniqueSkeeter** [comment]
async def update_product(param: UpdateProductParam):
    be_code = param.beCode
    if be_code is None:
        return UpdateProductResponse(result=False, resultMessage="header参数，beCode不能为空")
    if be_code != "CN":
        return UpdateProductResponse(result=False, resultMessage="header参数，beCode值不合法，必须为CN")
    try:
        collection_name = "tbl_hshop_product_kb"
        # 创建 Collection 对象
        collection = get_milvus_collection(collection_name)
        if param.type == 1:
            # 更新商品
            logger.info("update_product 更新商品开始")
            # 先删除已有的商品
            for product in param.productList:
                query_filter = f"display_prd_id=={product.disPlayPrdId}"
                delete_row = collection.delete(query_filter)
                logger.info(f'delete display_prd_id=={product.disPlayPrdId}, rows: {delete_row}')
            collection.flush()
            # 将商品信息转为Document
            metadata_list = to_product_kb_list(param.productList)
            # 文档向量化，并创建数据库
            add_rows = collection.insert(metadata_list)
            collection.flush()
            logger.info(f'add_rows: {add_rows}')

            # 更新商品数据入redis
            add_product_param(param.beCode, param.productList)
            logger.info("update_product 更新商品结束")
            return UpdateProductResponse(result=True, resultMessage="更新商品成功")
        elif param.type == 2:
            # 删除商品
            logger.info("update_product 删除商品开始")
            # 先删除已有的商品
            for product in param.productList:
                query_filter = f"display_prd_id=={product.disPlayPrdId}"
                delete_row = collection.delete(query_filter)
                logger.info(f'delete display_prd_id=={product.disPlayPrdId}, rows: {delete_row}')
            delete_product_param(param.beCode, param.productList)
            logger.info("update_product 删除商品结束")
            return UpdateProductResponse(result=True, resultMessage="删除商品成功")
        else:
            return UpdateProductResponse(result=False, resultMessage="type参数错误")
    finally:
        # 断开与 Milvus 的连接
        connections.disconnect(MILVUS_DEFAULT)

**Comment by uniqueSkeeter** [comment]
try:
        collection_name = "tbl_hshop_product_kb"
        # 创建 Collection 对象
        collection = get_milvus_collection(collection_name)
        if param.type == 1:
            # 更新商品
            logger.info("update_product 更新商品开始")
            # 先删除已有的商品
            for product in param.productList:
                query_filter = f"display_prd_id=={product.disPlayPrdId}"
                delete_row = collection.delete(query_filter)
                logger.info(f'delete display_prd_id=={product.disPlayPrdId}, rows: {delete_row}')
            collection.flush()
            # 将商品信息转为Document
            metadata_list = to_product_kb_list(param.productList)
            # 文档向量化，并创建数据库
            add_rows = collection.insert(metadata_list)
            collection.flush()
            logger.info(f'add_rows: {add_rows}')

            # 更新商品数据入redis
            add_product_param(param.beCode, param.productList)
            logger.info("update_product 更新商品结束")
            return UpdateProductResponse(result=True, resultMessage="更新商品成功")

**Comment by uniqueSkeeter** [comment]
i use "collection.flush()"，and it a liittle work but didn't work out completely

**Comment by xiaofan-luan** [comment]
use 2.4.19. We fix multiple bugs on delete.

If you still hit the same issue please let us know
